lib/tasks/application.rake

```diff
@@ -121,4 +123,45 @@ namespace :application do
     Rails.logger = ActiveSupport::Logger.new($stdout)
     Rails.logger.level = 1
   end
+
+  namespace :links_valid do
+    desc "Checks the validity of links in division summary"
+    task divisions: :environment do
+      md = Redcarpet::Markdown.new(Redcarpet::Render::HTML)
+
+      Division.find_each do |division|
+        wiki_motion = division.wiki_motion
+
+        if wiki_motion
+          summary_in_html = md.render(wiki_motion.description)
+          parsed_data = Nokogiri::HTML.parse(summary_in_html)
+          url_hash_table = Hash.new(0)
+          broken_urls = []
+
+          tags = parsed_data.xpath("//a")
+          tags.each do |tag|
+            url_from_page = tag[:href]
+            url = url_from_page
+
+            # adds http to the start of the URL to avoid failure since web request will fail without http://
+            url = "http://#{url}" if !url[%r{\Ahttp://}] && !url[%r{\Ahttps://}]
+
+            begin
+              if (url_hash_table[url]).zero?
```

Comment:
I can see how this will avoid doing multiple web requests when there a single url appears multiple times in a particular division. However, I don't think this will handle the (more likely) case where a particular url appears multiple times across many divisions. We want to make sure that even in that case we don't do multiple web requests to the same url.

Reply:
Working on this now.